datadog/dogstatsd/base.py

```diff
@@ -1066,5 +1148,28 @@ def _set_container_id(self, container_id, origin_detection_enabled):
                 log.debug("Couldn't get container ID: %s", str(e))
                 self._container_id = None
 
+    def _start_sender_thread(self):
+        log.debug("Starting background sender thread")
+        self._sender_thread = threading.Thread(
+            name="{}_sender_thread".format(self.__class__.__name__),
+            target=self._sender_main_loop,
+        )
+        self._sender_thread.daemon = True
+        self._sender_thread.start()
+
+    def _sender_main_loop(self):
+        while True:
+            self._xmit_packet_with_telemetry(self._queue.get())
+            self._queue.task_done()
+
+    def wait_for_pending(self):
```

Comment:
Should we consider documenting the new need (when using the threaded sender) to use this method to make sure everything's flushed properly and no buffered metrics are lost on shutdown?

Reply:
Good idea, done